cmd/spire-server/cli/run/run.go

```diff
@@ -480,8 +499,8 @@ func validateConfig(c *Config) error {
 		return errors.New("bind_address and bind_port must be configured")
 	}
 
-	if c.Server.RegistrationUDSPath == "" {
-		return errors.New("registration_uds_path must be configured")
```

Comment:
This check threw me off a bit as it made me think that this configurable is required (it doesn't seem to be ...?). The new check is different (presence vs mutual exclusion). I don't think it's a problem per se, but I thought I'd point it out.

Reply:
Previous to this, the RegistrationUDSPath was set to the default in the config structure before HCL was parsed, so at this point in the code it was required to be set. In order to test mutual exclusion of the options, I needed to know if SocketPath was explicitly set for the check so i opted to remove the default being set prior to HCL parsing, and test if both flags had a value here, and then apply the default later if neither was set.

There is a slight difference in behavior now. If you explicitly set the option to the empty string then we'd now apply the default instead of failing. If that is worrisome, we could change SocketPath to a `*string` to be able to distinguish between empty and unset.

Reply:
Ah ok... no I think it's alright 👍